### ISSUE TITLE
`null` response from server is returned as empty object

### ISSUE THREAD

**Issue by manoharank** [ember-cli/ember-ajax#158]
Upgrading to 2.5.0 from 0.7.1. #126 breaks my app, I expect a `null` value from the API, but now this returns an empty object, which breaks all my `{{#if data}}`. cc @alexlafroscia @taras

**Comment by taras** [comment]
@manoharank we made that change to ensure that payload always has a consistent errors object, but it looks like in the process we forced payload to become a hash. This is a bug.

I believe we should be doing something like this instead:

```diff
  handleResponse(status, headers, payload, requestData) {
-    payload = (payload === null || payload === undefined) ? {} : payload;
    const errors = this.normalizeErrorResponse(status, headers, payload);

    if (this.isSuccess(status, headers, payload)) {
      return payload;
```
and 

```diff
  normalizeErrorResponse(status, headers, payload) {
+ payload = isNone(payload) ? {} : payload;

    if (isArray(payload.errors)) {
```

This should address your problem. If @alexlafroscia agrees, would you be open to submitting a PR for this?

**Comment by manoharank** [comment]
@taras `Ember.isNone` returns `true` for `null`. We can have `payload = (payload === undefined) ? {} : payload; `.

**Comment by taras** [comment]
@manoharank `Ember.isNone(null) //=> true` will work as intended in the example that I showed above because we're not reassigning payload in `handleResponse`.

**Comment by manoharank** [comment]
@taras yes, just now looked the function name. And also I have raised one more bug related to `normalizeErrorResponse` #157 .

**Comment by dwickern** [comment]
If ember-ajax converts a `null` response (or an empty response with a 204 status code) into `{}`, ember-data will treat it as a record and expect it to have an `id`.

Ember-ajax should not convert a valid response into something else.

**Comment by XaserAcheron** [comment]
Confirmed -- pushed a failing test to my fork.

Interestingly, there was already a unit test responsible for checking if payloads remained unchanged, but the `null` and `undefined` cases were being compared to the empty object, which seems like the big "oops" here. After correcting the test, the issue was reproduced as described.

The fix suggested by @taras indeed makes everything pass again, BTW. :)

**Comment by alexlafroscia** [comment]
Closed by #232
